resources/frontend.rb

```diff
@@ -19,8 +19,19 @@
       variables['frontend'][new_resource.name] ||= {}
       variables['frontend'][new_resource.name]['default_backend'] ||= ''
       variables['frontend'][new_resource.name]['default_backend'] << new_resource.default_backend
-      variables['frontend'][new_resource.name]['bind'] ||= ''
-      variables['frontend'][new_resource.name]['bind'] << new_resource.bind
+      if new_resource.bind.is_a? Hash
+        variables['frontend'][new_resource.name]['bind'] = []
```

Comment:
should use `||=` instead of `=` otherwise you will loose any existing values.  Also should move it up right before the `if` and remove the duplicate line in the else.